staging/src/k8s.io/apiserver/pkg/registry/generic/rest/proxy.go

```diff
@@ -78,6 +78,8 @@ func NewUpgradeAwareProxyHandler(location *url.URL, transport http.RoundTripper,
 
 // ServeHTTP handles the proxy request
 func (h *UpgradeAwareProxyHandler) ServeHTTP(w http.ResponseWriter, req *http.Request) {
+	utilnet.PrepareForwardedFor(req)
```

Comment:
Question to reviewers:

Is it safe to modify the request headers here? Or do we need to copy the request first?